src/main/java/org/haffson/adventofcode/days/day01/Day01.java

```diff
@@ -1,32 +1,33 @@
 package org.haffson.adventofcode.days.day01;
 
+import org.haffson.adventofcode.utils.DataLoader;
 import org.haffson.adventofcode.ProblemStatusEnum;
 import org.haffson.adventofcode.days.Days;
+import org.haffson.adventofcode.utils.ProblemStatus;
 import org.springframework.stereotype.Component;
-import java.io.FileNotFoundException;
-import java.io.InputStream;
+
 import java.util.*;
+import java.util.stream.Collectors;
 
 /**
  * Implementation for <i>Day 1: Chronal Calibration</i>.
  */
 @Component
 public class Day01 implements Days {
 
-    /**
-     * The puzzle status {@code HashMap}
-     */
     private final Map<Integer, ProblemStatusEnum> problemStatus;
+    private final List<Integer> numbers;
 
-    // Read content of input file
-    public InputStream resource = getClass().getResourceAsStream("/data/day01/input_day01.txt");
-    private final Integer[] data = getRawData(resource);
+    public Day01(String filename) {
+        //get data
+        this.numbers = DataLoader.getDataDay01("/day01/" + filename, "\n");
+        // set ProblemStatus
+        this.problemStatus = ProblemStatus.getProblemStatusMap(1, 2,
```

Comment:
Hier ist das vielleicht erstmal egal, aber es ist ja so, dass man im Konstruktor schon weiß, dass beide Aufgaben gelöst sind. Man würde das in anderen Domänen/Projekten/etc. vielleicht eher so machen, dass am Ende eine Aufgabe der Status gesetzt wird.

Reply:
Keine schlechte Idee. 🤔